### PR TITLE
Extract both answer key parts, affects EBSR items

### PR DESCRIPTION
**Pull request by malaffoon**

### CODE REVIEW

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemService.java [hunk 1]
```diff
@@ -3,7 +3,6 @@
 import com.google.common.collect.ImmutableList;
```
Comment (malaffoon) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemService.java]: This is where all the real changes are:
* a little clean up to get rid of warnings, code duplication
* get both answer keys and join them with a semicolon (this matches the spec from tabulator output)
* hard-code name/desc fields for these items

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemService.java [hunk 2]
```diff
@@ -200,22 +213,4 @@ private String encodeImages(final String html, final String id) {
                 .collect(toImmutableList());
     }
 
-    private String safeValue(final Node node, final XPathExpression expression, final String defaultValue) {
```
Comment (malaffoon) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemService.java]: This was always being called with defaultValue="" so get rid of that param. And this method was a duplicate of the generic version so call it with the appropriate Function.

File: reporting-service/src/test/resources/items/Item-187-123/item-187-123.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (malaffoon) [inline reporting-service/src/test/resources/items/Item-187-123/item-187-123.xml]: A fake item file with both answer key values blank.

File: webapp/src/main/webapp/package-lock.json [hunk 4]
```diff
@@ -81,9 +81,9 @@
           "integrity": "sha512-4FG+RSG9DL7uEwRUZXZn3SS34DiDPfzP0VOiEwtUWlE+AR2EIg+hSyvrIgUUfhdgR/UkAeW2QHgeP+hWrXs7jQ==",
```
Comment (malaffoon) [inline webapp/src/main/webapp/package-lock.json]: Didn't mean to change or check this in.

File: reporting-service/src/test/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemServiceTest.java [hunk 5]
```diff
@@ -26,17 +25,18 @@
...
     public void before() throws Exception {
         final ArtifactRepository repository = mock(ArtifactRepository.class);
         when(repository.getItemXml(anyString())).thenAnswer(new Answer<InputStream>() {
             @Override
-            public InputStream answer(final InvocationOnMock invocation) throws Throwable {
+            public InputStream answer(final InvocationOnMock invocation) {
                 final String id = (String) invocation.getArguments()[0];
-                // TODO would like to reuse existing duplicate items under classpath:/items/...
-                return this.getClass().getResourceAsStream(MessageFormat.format("/{0}.xml", id));
+                return this.getClass().getResourceAsStream("/items/Item-" + id + "/item-" + id + ".xml");
```
Comment (malaffoon) [inline reporting-service/src/test/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemServiceTest.java]: Took care of this TODO, and that's what most of the test file changes below are all about.

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemService.java [hunk 6]
```diff
@@ -99,47 +101,54 @@ private synchronized Document getXmlDocument(final String id) throws IOException
...
+            return null;
+        }
+
+        final StringJoiner joiner = new StringJoiner(";");
+        if (answerKey != null) {
+            joiner.add(answerKey);
+        }
+        if (answerKey2 != null) {
+            joiner.add(answerKey2);
+        }
```
Comment (esotrope) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemService.java]: do we end up displaying `answer1;answer2`?

File: reporting-service/src/test/resources/items/Item-187-2638/item-187-2638.xml [hunk 7]
```diff
@@ -79,66 +79,66 @@
...
         Rubric 2</name>
-          <val><![CDATA[<p style="">&#xA0;</p>]]></val>
+          <val><![CDATA[<p style=""> </p>]]></val>
         </rubric>
         <samplelist maxval="2" minval="2">
           <sample purpose="Exemplar" scorepoint="2">
             <name>2-Point Official Sample Answer
       </name>
             <annotation />
-            <samplecontent><![CDATA[<p style="">&#xA0;</p>]]></samplecontent>
```
Comment (esotrope) [inline reporting-service/src/test/resources/items/Item-187-2638/item-187-2638.xml]: what character is this we are replacing here?

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemService.java (hunk 6 above)
Comment (malaffoon) [reply reporting-service/src/main/java/org/opentestsystem/rdw/reporting/item/DefaultExamItemService.java]: Correct. It looks something like "C;B". Similar to MS answer key which is like "A,B,D". The choice of semicolon was to match the tabulator output format.

File: reporting-service/src/test/resources/items/Item-187-2638/item-187-2638.xml (hunk 7 above)
Comment (malaffoon) [reply reporting-service/src/test/resources/items/Item-187-2638/item-187-2638.xml]: Some sort of hard space. I could've changed the test but it seemed easier to just get rid of the cruft.

**Pull request merged**